hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRPCClientRetries.java

```diff
@@ -155,10 +155,10 @@ public void testRetryWhenOneNameServiceDown() throws Exception {
     // Renew lease for the DFS client, it will succeed.
     routerProtocol.renewLease(client.getClientName());
 
-    // Verify the retry times, it will retry one time for ns0.
+    // Verify the retry times, it should succeed with no retry as long as at least one of the nameservices is ACTIVE.
     FederationRPCMetrics rpcMetrics = routerContext.getRouter()
         .getRpcServer().getRPCMetrics();
-    assertEquals(1, rpcMetrics.getProxyOpRetries());
+    assertEquals(0, rpcMetrics.getProxyOpRetries());
```

Comment:
With this patch, ns0 is no longer a target namespace of the renewLease RPC and thus there should be no retries.
https://github.com/apache/hadoop/blob/02d578a9a1d2e084b609c24640c51a783a598464/hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java#L595-L596

Reply:
Somehow the unit test without this fix passed in trunk. I had an offline discussion with @hdaikoku, but we still don't know the cause. We will create an issue if necessary.